store/pruning.go

```diff
@@ -1,24 +1,23 @@
 package store
 
 import (
+	"errors"
 	"fmt"
 
-	"github.com/dymensionxyz/gerr-cosmos/gerrc"
-
 	"github.com/dymensionxyz/dymint/types"
+	"github.com/dymensionxyz/gerr-cosmos/gerrc"
 )
 
-// PruneStore removes store items up to (but not including) a height. It returns number of blocks pruned.
-func (s *DefaultStore) PruneStore(from, to uint64, logger types.Logger) (uint64, error) {
-	if from <= 0 {
-		return 0, fmt.Errorf("from height must be greater than 0: %w", gerrc.ErrInvalidArgument)
-	}
-
-	if to <= from {
-		return 0, fmt.Errorf("to height must be greater than from height: to: %d: from: %d: %w", to, from, gerrc.ErrInvalidArgument)
+// PruneStore removes blocks up to (but not including) a height. It returns number of blocks pruned.
+func (s *DefaultStore) PruneStore(to uint64, logger types.Logger) (uint64, error) {
+	prunedBlocks := uint64(0)
+	from, err := s.LoadBaseHeight()
+	if errors.Is(err, gerrc.ErrNotFound) {
+		logger.Error("load store base height", "err", err)
+	} else if err != nil {
+		return prunedBlocks, err
 	}
-
-	prunedBlocks, err := s.pruneBlocks(from, to, logger)
+	prunedBlocks, err = s.pruneBlocks(from, to, logger)
```

Comment:
if we e.g get erorr here, than we still save the `to` as base height. (i.e some stores succeeded and some not).
not sure if critical for now but probably worth opening an issue on it otherwise you have stores which are partially pruned.

Reply:
i actually refactored the func to prune all indexes for every height iteration, so in case it fails it will be more likely all failed at the same height, although the solution not perfect. i created issue for this https://github.com/dymensionxyz/dymint/issues/1196